master/internal/stream/system.go

```diff
@@ -255,72 +254,62 @@ func (ps *PublisherSet) Websocket(
 		}
 	}()
 
-	// reads is where we collect SubscriptionModMsg messages we read from the websocket until
-	// waitForSomething() delivers those messages to the websocket goroutine.
-	var reads []SubscriptionModMsg
-
-	// always be reading for new subscriptions
+	// monitor for new StartupMsgs
 	go func() {
-		defer streamer.Close()
 		for {
-			var mods SubscriptionModMsg
-			err := socket.ReadJSON(&mods)
+			select {
+			// check if we're canceled
+			case <-ctx.Done():
+				return
+			default:
+			}
```

Comment:
Why add this, when we're blocking in other parts of this goroutine?

Reply:
This is an artifact of trying to identify the best way to handle `ReadJSON()` failing due to the websocket being closed. It should be removed